src/posit/connect/cursors.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, make_dataclass
+from typing import Generator, List
+
+import requests
+
+# The maximum page size supported by the API.
+_MAX_PAGE_SIZE = 500
+
+
+@dataclass
+class Page:
+    paging: dict
+    results: List[dict]
+
+
+class Paginator:
```

Comment:
Is it confusing that we have two classes called Paginator, just in different submodules? Would it make sense to name these differently?